tests/d3d12_mesh_shader.c

```diff
@@ -0,0 +1,2326 @@
+    { {
+        D3D12_PIPELINE_STATE_SUBOBJECT_TYPE_INPUT_LAYOUT,
+        { NULL, 0 },
+    } };
+
+    static const union d3d12_primitive_topology_subobject primitive_topology_subobject =
+    {{
+        D3D12_PIPELINE_STATE_SUBOBJECT_TYPE_PRIMITIVE_TOPOLOGY,
+        D3D12_PRIMITIVE_TOPOLOGY_TYPE_TRIANGLE,
+    }};
+
+    static const union d3d12_rasterizer_subobject rasterizer_subobject =
+    {{
+        D3D12_PIPELINE_STATE_SUBOBJECT_TYPE_RASTERIZER,
+        { D3D12_FILL_MODE_SOLID, D3D12_CULL_MODE_BACK,
+            FALSE, 0, 0.0f, 0.0f, TRUE, FALSE, FALSE, 0,
+            D3D12_CONSERVATIVE_RASTERIZATION_MODE_OFF },
+    }};
+
+    static const union d3d12_sample_desc_subobject sample_desc_subobject =
+    {{
+        D3D12_PIPELINE_STATE_SUBOBJECT_TYPE_SAMPLE_DESC,
+        { 1, 0 },
+    }};
+
+    struct {
+        union d3d12_root_signature_subobject root_signature;
+        union d3d12_shader_bytecode_subobject ms;
+    } ms_only_pipeline_desc;
+
+    struct {
+        union d3d12_root_signature_subobject root_signature;
+        union d3d12_shader_bytecode_subobject ms;
+        union d3d12_shader_bytecode_subobject vs;
+        union d3d12_primitive_topology_subobject primitive_topology;
+        union d3d12_input_layout_subobject input_layout;
+        union d3d12_rasterizer_subobject rasterizer;
+    } ms_vs_pipeline_desc;
+
+    struct {
+        union d3d12_root_signature_subobject root_signature;
+        union d3d12_shader_bytecode_subobject ms;
+        union d3d12_shader_bytecode_subobject ps;
+        union d3d12_sample_desc_subobject sample_desc;
+        union d3d12_rasterizer_subobject rasterizer;
+    } ms_ps_pipeline_desc;
+
+    struct {
+        union d3d12_root_signature_subobject root_signature;
+        union d3d12_shader_bytecode_subobject ms;
+        union d3d12_primitive_topology_subobject primitive_topology;
+        union d3d12_input_layout_subobject input_layout;
+    } ms_ia_pipeline_desc;
+
+    if (!(device = create_device()))
+    {
+        skip("Failed to create device.\n");
+        return;
+    }
+
+    memset(&options7, 0, sizeof(options7));
+    hr = ID3D12Device_CheckFeatureSupport(device, D3D12_FEATURE_D3D12_OPTIONS7, &options7, sizeof(options7));
+    ok(SUCCEEDED(hr), "OPTIONS7 is not supported by runtime.\n");
+
+    if (options7.MeshShaderTier < D3D12_MESH_SHADER_TIER_1)
+    {
+        skip("Mesh shaders not supported by device.\n");
+        ID3D12Device_Release(device);
+        return;
+    }
+
+    hr = ID3D12Device_QueryInterface(device, &IID_ID3D12Device2, (void **)&device2);
+    ok(SUCCEEDED(hr), "Failed to query ID3D12Device2.\n");
+
+    root_signature_desc.NumParameters = 0;
+    root_signature_desc.pParameters = NULL;
+    root_signature_desc.NumStaticSamplers = 0;
+    root_signature_desc.pStaticSamplers = NULL;
+    root_signature_desc.Flags = D3D12_ROOT_SIGNATURE_FLAG_ALLOW_INPUT_ASSEMBLER_INPUT_LAYOUT;
```

Comment:
Is this required for mesh shaders? I thought input layout was just VS.

Reply:
It's not, the point here is that pipeline compilation still succeeds and just ignores these subobjects.

Reply:
OK, fair enough.